content/WebAuthn/WebAuthn_Developer_Guide/webauthn-readiness-checklist.adoc

```diff
@@ -0,0 +1,198 @@
+
+* We recommend your solution not limit access based on authenticator model or attestation statement, unless required by your use case.
+* We recommend your solution not limit access based on the user-agent value reported by the browser.
+
+
+== Authenticator Lifecycle
+
+* Users must be able to register at least two security keys per account (one primary and one backup)
+* Users must be able to name or rename a registered security key
+* Users should be presented with the date and time when the key was last used
+* Users must be able to remove a registered security key from their account
+* Administrators should be able to remove a registered security key on behalf of a user if authorized.
+* Users should be able to configure a range of other account recovery options if all security keys are lost (e.g., backup codes, etc…)
+* Provide instructions to inform users how to register, authenticate, and remove security keys.
+
+
+== Testing Your Integration
+
+Test with your users’ browsers and devices.
+
+
+=== Stability
+
+The solution must not freeze, crash, rapidly drain battery, or put unnecessary strain on device resources.
+
+
+=== User Experience
+
+The solution must communicate YubiKey status to users. This section does not apply if using a FIDO2/WebAuthn-compatible browser.
+
+* Display an animation to indicate that the security key should be inserted or guide the user to the appropriate location for NFC.
+* Display an animation to prompt the user to take action on the security key.
+
+
+=== Branding
+
+Your solution must follow the https://www.yubico.com/press/images/[Yubico usage guidelines] when presenting the Yubico image or logo.
+
+
+=== Functional Testing
+
+Perform all the following tests. Before performing each test, enable or disable as required the specified YubiKey functionality. For this, the https://developers.yubico.com/yubikey-manager/[YubiKey Manager] might be necessary in order to enable/disable specific functionality of your YubiKey.
+
+* Register a YubiKey (only CTAP1/U2F enabled)
+* Register a YubiKey (only CTAP2/FIDO2 enabled)
+* Register a second YubiKey (only CTAP1/U2F enabled)
+* Register a second YubiKey (only CTAP2/FIDO2 enabled)
+* Prompt to insert YubiKey as appropriate for registration
+* Prompt to touch YubiKey as appropriate for registration
+* Gracefully recover if a YubiKey is not present for registration
+* Gracefully recover if a YubiKey is not touched for registration
+* Authenticate using YubiKey (CTAP1/U2F enabled)
+* Authenticate using YubiKey (CTAP2/FIDO2 enabled)
+* Prompt to insert YubiKey as appropriate for authentication
+* Prompt to touch YubiKey as appropriate for authentication
+* Gracefully recover if a YubiKey is not presented for authentication
+* Gracefully recover if a YubiKey is not touched for authentication
+* Unregister/remove a YubiKey (self-service or admin request)
+* Name or rename registered YubiKey
+* Prevent the same user from registering the same YubiKey multiple times
```

Comment:
Stray thought: is it sufficiently clear that this is about having the same YubiKey registered with multiple credentials at the same time - as opposed to having it registered, then de-registering it and then registering it again?

Reply:
No, I don't think it is clear enough. Please will you change it to read "* Prevent a user from registering the same YubiKey with multiple credentials at the same time, as opposed to registering the YubiKey, deregistering it, then re-registering it ad infinitum, which is permissible."

Reply:
Opened meta-pull-request: FFincannon/developers.yubico.com#1